zstd_test.go

```diff
@@ -162,6 +194,25 @@ func TestRealPayload(t *testing.T) {
 	}
 }
 
+func TestRealPayloadDict(t *testing.T) {
+	dict, err := ioutil.ReadFile("test_dictionary")
+	if err != nil {
+		t.Fatalf("Error getting dictionary: %v", err)
+	}
+
+	dst, err := CompressLevelDict(nil, raw, DefaultCompression, dict)
+	if err != nil {
+		t.Fatalf("Failed to compress: %s", err)
+	}
+	rein, err := DecompressDict(nil, dst, dict)
+	if err != nil {
+		t.Fatalf("Failed to decompress: %s", err)
+	}
+	if string(raw) != string(rein) {
+		t.Fatalf("compressed/decompressed payloads are not the same (lengths: %v & %v)", len(raw), len(rein))
+	}
+}
+
```

Comment:
Do you mind also adding benchmarks `BenchmarkCompressionDict` & `BenchmarkDecompressionDict` so we can catch any future performance regressions on those APIs ?